File: strategy-game-backend/src/main/kotlin/de/ruegnerlukas/strategygame/backend/ports/models/dtos/ProductionQueueEntryDTO.kt
package de.ruegnerlukas.strategygame.backend.ports.models.dtos

import de.ruegnerlukas.strategygame.backend.ports.models.BuildingType

data class ProductionQueueEntryDTO(
    val entryId: String,
    val buildingType: BuildingType,
    val progress: Float
)

File: frontend/src/external/api/models/productionQueueEntryDTO.ts
export interface ProductionQueueEntryDTO {
    type: "building" | "settler"
    entryId: string,
    progress: number
}

export interface BuildingProductionQueueEntryDTO extends ProductionQueueEntryDTO {
    buildingType: string;
}

export interface SettlerProductionQueueEntryDTO extends ProductionQueueEntryDTO {
}